cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -262,7 +318,8 @@ public void update(Id.DatasetInstance instance, Map<String, String> properties)
    * @throws DatasetNotFoundException if the dataset instance was not found
    * @throws IOException if there was a problem in checking if the namespace exists over HTTP
    */
-  public void drop(Id.DatasetInstance instance) throws Exception {
+  void drop(Id.DatasetInstance instance) throws Exception {
+    authorizationEnforcementService.enforce(instance.toEntityId(), SecurityRequestContext.toPrincipal(), Action.ADMIN);
```

Comment:
The reason for that pattern is that you are creating a new entity, so the privileges are required on the parent. The reason for choosing "write" access was that this operation is like "adding" something to the parent, not "updating" an existing thing, which will require "admin"